ratis-server/src/main/java/org/apache/ratis/server/raftlog/segmented/SegmentedRaftLogWorker.java

```diff
@@ -550,6 +556,7 @@ public void execute() throws IOException {
         LOG.info("{}: Deleted empty log segment {}", name, openFile);
       }
       updateFlushedIndexIncreasingly();
+      updateSafeCacheEvitIndex(endIndex);
```

Comment:
Also should this be updateIncreasingly ? as when the system is running then this index should be updated increasingly.

Reply:
We call FinalizeLogSegment after a snapshot is installed too. This is done so that open segments with endIndex less than snapshotIndex can be closed. During this FinalizeLogSegment, updateIncreasingly will through a Precondition error as safeCacheEvictIndex would be updated to the snapshotIndex already.

Should I separate the two FinalizeLogSegment calls using some flag?